src/core/algorithms/algo_factory.cpp

```diff
@@ -47,7 +48,9 @@ void LoadAlgorithm(Algorithm& algorithm, StdParamsMap const& options) {
     ConfigureFromFunction(algorithm, [&options](std::string_view option_name) {
         using namespace config::names;
         auto create_input_table = [](CSVConfig const& csv_config) -> config::InputTable {
-            return std::make_shared<CSVParser>(csv_config);
+            auto csv_parser = std::make_shared<CSVParser>(csv_config);
+            csv_parser->ValidateSeparator();
+            return csv_parser;
```

Comment:
Also, I'm not sure that we always want to validate the separator when creating a table. The main problem is that we have to read the table completely before running the algorithm itself. This causes the file system caches and disk cache to fill up. Therefore, if someone tries to conduct experiments and does not know that validation always takes place first, then the results of the experiments will be incorrect (since all caches must be reset before running the test).